py/client/pydeephaven/session.py

```diff
@@ -53,31 +58,23 @@ def call_completed(self, exception):
 
     def received_headers(self, headers):
         super().received_headers(headers)
-        if headers:
-            auth_token = bytes(headers.get("authorization")[0], encoding='ascii')
-            if auth_token and auth_token != self._session._auth_token:
-                self._session._auth_token = auth_token
+        header_key = "authorization"
+        try:
+            if headers and header_key in headers:
+                header_value = headers.get(header_key)
+                auth_header_value = bytes(header_value[0], encoding='ascii')
+                if auth_header_value:
+                    self._session._auth_header_value = auth_header_value
+        except Exception as e:
+            logger.exception(f'_DhClientAuthMiddleware.received_headers got headers={headers}')
+            return
 
     def sending_headers(self):
-        return {
-            **{
-                "authorization": self._session._auth_token
-            }, **self._session._extra_headers
-        }
-
+        return None
 
-class _DhClientAuthHandler(ClientAuthHandler):
-    def __init__(self, session):
-        super().__init__()
-        self._session = session
-        self._token = b''
 
-    def authenticate(self, outgoing, incoming):
-        outgoing.write(self._session._auth_token)
-        self._token = incoming.read()
-
-    def get_token(self):
-        return self._token
+def trace(who):
```

Comment:
1. If this is not intended to be public, it should start with `_`.
2. If this is intended to be public, it needs pydocs.
3. Missing type hints.

Reply:
Done